packages/site/cms-site/src/blocks/factories/SeoBlock.tsx

```diff
@@ -0,0 +1,71 @@
+import Head from "next/head";
+import * as React from "react";
+
+import { PixelImageBlockData, SeoBlockData } from "../../blocks.generated";
+import { withPreview, WithPreviewProps } from "../../iframebridge/withPreview";
+import { generateImageUrl } from "../../image/Image";
+import { PropsWithData } from "../PropsWithData";
+
+type SeoBlockProps<T = PixelImageBlockData> = (T extends PixelImageBlockData
+    ? PropsWithData<SeoBlockData> & { resolveOpenGraphImageUrlTemplate?: never }
+    : { resolveOpenGraphImageUrlTemplate: (block: T) => string | undefined } & PropsWithData<
+          Omit<SeoBlockData, "openGraphImage"> & { openGraphImage: { block?: T } }
+      >) & { title: string; canonicalUrl?: string };
+
+export const SeoBlock: React.FunctionComponent<WithPreviewProps & SeoBlockProps> = withPreview<SeoBlockProps>(
+    <T = PixelImageBlockData,>({
+        data: {
+            htmlTitle,
+            metaDescription,
+            openGraphTitle,
+            openGraphDescription,
+            openGraphImage,
+            noIndex,
+            canonicalUrl,
+            structuredData,
+            alternativeLinks,
+        },
+        title,
+        canonicalUrl: passedCanonicalUrl,
+        resolveOpenGraphImageUrlTemplate = (block) => (block as PixelImageBlockData).urlTemplate,
+    }: SeoBlockProps<T>) => {
+        const usedHtmlTitle = htmlTitle || title;
+        const usedCanonicalUrl = canonicalUrl ?? passedCanonicalUrl;
+        const openGraphImageUrlTemplate = openGraphImage.block && resolveOpenGraphImageUrlTemplate(openGraphImage.block as T);
+
+        return (
+            <>
```

Comment:
@johnnyomair Should I remove this fragment or is it necessary?

Reply:
Doesn't seem necessary to me, we may remove it